frontend/src/component/ui/Toggle.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { Component } from 'react';
+
+import './Toggle.scss';
+import classNames from 'classnames';
+
+interface Props {
+    onClick?: (enabled: boolean) => void;
+}
+
+interface State {
+    enabled: boolean,
+    toggleClassName: string,
+    toggleBackClassName: string
+}
+
+export class Toggle extends Component<Props, State> {
+    
+    constructor(props: Props) {
+        super(props);
+        this.state = {
+            enabled: false,
+            toggleClassName: classNames('toggle'),
+            toggleBackClassName: classNames('toggle-back')
+        };
+    }
+
+    handleClick = () => {
+        this.setState({
+            enabled: !this.state.enabled
+        }, () => {
+            const { enabled } = this.state;
+            if (!this.props.onClick) {
+                return;
+            }
+            this.props.onClick(enabled);
+            if (enabled) {
+                this.setState({
+                    toggleClassName: classNames('toggle', 'active'),
+                    toggleBackClassName: classNames('toggle-back', 'active')
+                });
+            } else {
+                this.setState({
+                    toggleClassName: classNames('toggle'),
+                    toggleBackClassName: classNames('toggle-back')
+                });
+            }
+        });
+    }
+
+    render() {
+        return <div className={'toggle-component'}>
+            <p className={'toggle-label'}>{this.props.children}</p>
```

Comment:
Aha, toggle is toggle. Let's separate toggle with label so that we can customize labels in the component using toggle.

Reply:
I removed the toggle label element, let me know if anything else should be done or if this is now resolved.